lightwood/encoders/time_series/helpers/rnn_helpers.py

```diff
@@ -28,17 +24,24 @@ def tensor_from_series(series, device, n_dims, pad_value, max_len=None, normaliz
             series[i].append(pad_value)
         series[i] = series[i][:max_len]
 
-    # dimension padding
-    for _ in range(max(0, n_dims - len(series))):
-        series.append([pad_value] * max_len)
-
-    # normalize and transpose
     if normalizer:
-        tensor = torch.Tensor([normalizer.encode(s) for s in series][0])
+        if isinstance(normalizer, MinMaxNormalizer):
+            tensor = torch.Tensor([normalizer.encode(s) for s in series]).transpose(0, 1)
+        else:
+            tensor = torch.Tensor([normalizer.encode(s) for s in series][0])
+
+        if len(tensor.shape) > 2:
+            tensor = tensor.squeeze(2)
     else:
-        tensor = torch.transpose(torch.Tensor(series), 0, 1)
+        tensor = torch.Tensor(series).transpose(0, 1)
```

Comment:
`.transpose(0, 1)` seems to be called once here, in some situation ,and once above for the MinMaxNormalizer.

Again, I'm not sure what the exact logic is but it seems to me like this second else/if could be made part of the one above ? Maybe not, just a thought.